File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/test/java/simulacionBandaMusica/utils/funcionesMenuTest.java
package simulacionBandaMusica.utils;

import org.junit.jupiter.api.Test;
import simulacionBandaMusica.factories.*;
import simulacionBandaMusica.models.Musico;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static simulacionBandaMusica.utils.funcionesMenu.calcularMantenimientoBanda;

class funcionesMenuTest {

    @Test
    void calcularMantenimientoBandaTest() {
        Musico[] banda = {
                BajistaFactory.create(), CantanteFactory.create(),
                CantanteProFactory.create(), GuitarristaFactory.create(),
                MultiFactory.create(), PercusionistaFactory.create(),
                TeclistaFactory.create(), TrompetistaFactory.create()
        };
        
        assertEquals(16875, calcularMantenimientoBanda(banda));
    }
}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/test/kotlin/simulacionBandaMusica/utils/funcionesMenuTest.kt
package simulacionBandaMusica.utils

import org.junit.jupiter.api.Test
import simulacionBandaMusica.factories.*
import simulacionBandaMusica.models.Musico
import simulacionBandaMusica.utils.funcionesMenu.calcularMantenimientoBanda
import kotlin.test.assertEquals

internal class funcionesMenuTest {
    @Test
    fun calcularMantenimientoBandaTest() {
        val banda = arrayOfNulls<Musico>(8)
        val almacenMusicos = arrayOf(
            BajistaFactory.create(), CantanteFactory.create(),
            CantanteProFactory.create(), GuitarristaFactory.create(),
            MultiFactory.create(), PercusionistaFactory.create(),
            TeclistaFactory.create(), TrompetistaFactory.create()
        )

        for (i in 0 until banda.size) {
            banda[i] = almacenMusicos[i]
        }
        
        assertEquals(16875, calcularMantenimientoBanda(banda))
    }
}